diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -132,7 +132,7 @@
 # rather than finding our external thirdparty libraries first
 find_package( ZLIB   REQUIRED )
 find_package( PNG    REQUIRED )
-find_package( Jasper 1.900.1...1.900.29 REQUIRED )
+find_package( Jasper 1.900.1 REQUIRED )
 
 # Because WRF might use a different version of zlib, check to see if we are using a different one
 # when externals (really internals) are used
